api/views.py

```diff
@@ -419,3 +502,46 @@ def approve(self, request, pk):
         winery.available_since = datetime.now()
         winery.save()
         return Response({'detail': 'Winery successfully approved'}, status=status.HTTP_200_OK)
+
+
+class ReportsView(APIView):
+    def get(self, request, *args, **kwargs):
+        user_events = request.user.winery.events.all()
+        response = {
+            "reservations_by_event": (
+                Reservation.objects
+                .values("event_occurrence__event__name")
+                .annotate(name=F("event_occurrence__event__name"))
+                .annotate(count=Count("id"))
+                .values("name", "count")
+                .filter(event_occurrence__event__in=user_events)
+                .order_by("count")[:10]
+            ),
+            "reservations_by_month": (
+                Reservation.objects.annotate(month=ExtractMonth("event_occurrence__start"))
+                .values("month")
+                .annotate(count=Count("id"))
+                .values("month", "count")
+                .order_by("month")
+            ),
+            "attendees_languages": (
+                Reservation.objects
+                .values("user__language")
+                .annotate(language=F("user__language"))
+                .annotate(count=Count("id"))
+                .values("language", "count")
+                .filter(event_occurrence__event__in=user_events)
+            ),
+        }
+
+        # Awful hack to return months with count = 0
+        # To achieve this with SQL, a Month table is needed :(
+        zero_count_months = [{"month": i, "count": 0} for i in range(1, 13)]
```

Comment:
Doesn't  chart js solve this for us?. It's ok though

Reply:
Sadly, it doesn't. :(